prefect_great_expectations/validation.py

```diff
@@ -0,0 +1,102 @@
+            "Check result on this exception for more details."
+        )
+
+
+@task
+def run_checkpoint_validation(
+    run_name: Optional[str] = None,
+    checkpoint_name: Optional[str] = None,
+    checkpoint: Optional[Checkpoint] = None,
+    checkpoint_kwargs: Optional[Dict] = None,
+    data_context_root_dir: Optional[Union[str, Path]] = None,
+    data_context: Optional[DataContext] = None,
+    runtime_environment: Optional[Dict] = None,
+    raise_on_validation_failure: bool = True,
+):
+    """
+    Task that performs a Great Expectations validation based on the provided checkpoint
+        and data context
+
+    Args:
+        run_name: The name of the Great Expectations validation run. Defaults to
+            timestamp if not provided.
+        checkpoint_name: The name of the Checkpoint to use for validation.
+        checkpoint: A Checkpoint object to use for validation. Overrides
+            `checkpoint_name` if both are provided.
+        checkpoint_kwargs:  A dictionary with values used to provide configuration to
+            the task's Checkpint at runtime. Keys should match the parameters of
+            `CheckpointConfig`.
+        data_context_root_dir: Path to the great_expectations directory.
+        data_context: A DataContext object to use during validation. Overrides
+            `data_context_root_dir` if both are provided.
+        runtime_environment: A dictionary with values to overwrite config in
+            `great_expectations.yml` at run time.
+        raise_on_validation_failure: If `True`, the task will raise a
+            GreatExpectationValidationError when validation fails. If `False`,
+            the task will return the result of the validation.
+
+    Raises:
+        GreatExpectationValidationError: Signals that a GE validation failed.
+            Details of the failure can be found by inspecting the `result`
+            attribute of the exception.
+
+    Returns:
+        CheckpointResult: Detailed result of the validation run in the task.
+    """
+    logger = get_run_logger()
+
+    logger.info("Running Great Expectations validation...")
+
+    runtime_environment = runtime_environment or dict()
+    checkpoint_kwargs = checkpoint_kwargs or dict()
+
+    data_context_root_dir = (
+        str(data_context_root_dir) if data_context_root_dir else None
+    )
+
+    if data_context:
+        logger.debug("Using provided GE Data Context")
+    else:
+        logger.debug("Loading GE Data Context from %s", data_context_root_dir)
+        data_context = DataContext(
+            context_root_dir=data_context_root_dir,
+            runtime_environment=runtime_environment,
+        )
+
+    if checkpoint:
+        logger.debug("Using provided GE Checkpoint")
+    else:
+        logger.debug("Loading GE Checkpoint with name %s", checkpoint_name)
+        checkpoint = data_context.get_checkpoint(checkpoint_name)
+
+    result = checkpoint.run(run_name=run_name, **checkpoint_kwargs)
```

Comment:
So I assume checkpoint will automatically convert` run_name` to the default time?

Reply:
Yes, if `run_name` is `None` then GE uses a default for run identification.